front/src/modules/powerRestriction/helpers/powerRestrictionWarnings.ts

```diff
@@ -0,0 +1,96 @@
+import type { RangedValue, RollingStock } from 'common/api/osrdEditoastApi';
+import { NO_POWER_RESTRICTION } from 'modules/powerRestriction/consts';
+import type { PowerRestrictionWarnings } from 'modules/powerRestriction/types';
+import { getRollingStockPowerRestrictionsByMode } from 'modules/rollingStock/helpers/powerRestrictions';
+
+const getInvalidZoneBoundaries = (
+  powerRestrictionRange: RangedValue,
+  electrificationRange: RangedValue
+) => {
+  const invalidZoneBegin = Math.round(
+    electrificationRange.begin < powerRestrictionRange.begin
+      ? powerRestrictionRange.begin
+      : electrificationRange.begin
+  );
+  const invalidZoneEnd = Math.round(
+    powerRestrictionRange.end < electrificationRange.end
+      ? powerRestrictionRange.end
+      : electrificationRange.end
+  );
+  return { begin: invalidZoneBegin, end: invalidZoneEnd };
+};
+
+/**
+ * Depending on the electrification on a train's path and the power restriction codes selected by the user,
+ * returns the possible warnings based on the incompatibilty between them on each interval
+ * @returns A dictionnary of warnings ordered by power restriction codes
+ */
+export const getPowerRestrictionsWarnings = (
+  powerRestrictionRanges: RangedValue[],
+  pathElectrificationRanges: RangedValue[],
+  rollingStockModes: RollingStock['effort_curves']['modes']
+): PowerRestrictionWarnings => {
+  const powerRestrictionsByMode = getRollingStockPowerRestrictionsByMode(rollingStockModes);
+  const invalidCombinationWarnings: PowerRestrictionWarnings['invalidCombinationWarnings'] = [];
+  const modeNotSupportedWarnings: PowerRestrictionWarnings['modeNotSupportedWarnings'] = [];
+  const missingPowerRestrictionWarnings: PowerRestrictionWarnings['missingPowerRestrictionWarnings'] =
+    [];
```

Comment:
```
const powerRestrictionsByMode = getRollingStockPowerRestrictionsByMode(rollingStockModes);
  const warnings: PowerRestrictionWarnings = {
    invalidCombinationWarnings: [],
    modeNotSupportedWarnings: [],
    missingPowerRestrictionWarnings: []
  };
```

il would go for something like this to avoid repetition

Reply:
Done, it's clearer, thanks :pray: